homeassistant/components/leafspy/device_tracker.py

```diff
@@ -0,0 +1,206 @@
+            dev_id
+        )
+        entities.append(entity)
+
+    async_add_entities(entities)
+
+    return True
+
+
+class LeafSpyEntity(DeviceTrackerEntity, RestoreEntity):
+    """Represent a tracked car."""
+
+    def __init__(self, dev_id, data=None):
+        """Set up LeafSpy entity."""
+        self._dev_id = dev_id
+        self._data = data or {}
+        self.entity_id = ENTITY_ID_FORMAT.format(dev_id)
+
+    @property
+    def unique_id(self):
+        """Return the unique ID."""
+        return self._dev_id
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the car."""
+        return self._data.get('battery')
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific attributes."""
+        return self._data.get('attributes')
+
+    @property
+    def latitude(self):
+        """Return latitude value of the car."""
+        # Check with "get" instead of "in" because value can be None
+        if self._data.get('gps'):
+            return self._data['gps'][0]
+
+        return None
+
+    @property
+    def longitude(self):
+        """Return longitude value of the car."""
+        # Check with "get" instead of "in" because value can be None
+        if self._data.get('gps'):
+            return self._data['gps'][1]
+
+        return None
+
+    @property
+    def name(self):
+        """Return the name of the car."""
+        return self._data.get('host_name')
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def source_type(self):
+        """Return the source type of the car."""
+        return SOURCE_TYPE_GPS
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            'name': self.name,
+            'identifiers': {(LS_DOMAIN, self._dev_id)},
+        }
+
+    async def async_added_to_hass(self):
+        """Call when entity about to be added to Home Assistant."""
+        await super().async_added_to_hass()
+
+        # Don't restore if we got set up with data.
+        if self._data:
+            return
+
+        state = await self.async_get_last_state()
+
+        if state is None:
+            return
+
+        attr = state.attributes
+        self._data = {
+            'host_name': state.name,
+            'gps': (attr.get(ATTR_LATITUDE), attr.get(ATTR_LONGITUDE)),
+            'battery': attr.get(ATTR_BATTERY_LEVEL),
+            'source_type': attr.get(ATTR_SOURCE_TYPE),
+        }
+
+    @callback
+    def update_data(self, data):
+        """Mark the device as seen."""
+        self._data = data
+        self.async_write_ha_state()
+
+
+def _parse_see_args(message):
+    """Parse the Leaf Spy parameters, into the format see expects."""
+    dev_id = slugify('leaf_{}'.format(message['VIN']))
+    args = {
+        'dev_id': dev_id,
+        'host_name': message['user'],
```

Comment:
Why do we call this `host_name`? `device_name` matches better I think.

Reply:
Same as above